database/src/main/java/com/firebase/ui/database/FirebaseRecyclerAdapter.java

```diff
@@ -70,11 +70,10 @@
  * @param <VH> The ViewHolder class that contains the Views in the layout that is shown for each object.
  */
 public abstract class FirebaseRecyclerAdapter<T, VH extends RecyclerView.ViewHolder> extends RecyclerView.Adapter<VH> {
-
-    Class<T> mModelClass;
+    protected Class<T> mModelClass;
     protected int mModelLayout;
-    Class<VH> mViewHolderClass;
-    FirebaseArray mSnapshots;
+    protected Class<VH> mViewHolderClass;
+    protected FirebaseArray mSnapshots;
```

Comment:
Same as for the other adapter: the fields should not be accessible to subclasses, as those might otherwise modify them. The pertinent information from the fields should already be available through getters.